web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/ThumbnailController.java

```diff
@@ -112,13 +114,42 @@ public ResponseEntity<Object> assignThumbnail(@PathVariable("pidString") String
         var agent = AgentPrincipalsImpl.createFromThread();
         var request = new ThumbnailRequest();
         request.setAgent(agent);
-        request.setFilePidString(pidString);
+        request.setPidString(pidString);
         try {
             thumbnailRequestSender.sendToQueue(request);
         } catch (IOException e) {
-            log.error("Error assigning file {} as thumbnail", request.getFilePidString(), e);
+            log.error("Error assigning file {} as thumbnail", request.getPidString(), e);
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
         return new ResponseEntity<>(HttpStatus.OK);
     }
+
+    @DeleteMapping(value = "/edit/deleteThumbnail/{pidString}")
+    @ResponseBody
+    public ResponseEntity<Object> deleteThumbnail(@PathVariable("pidString") String pidString) {
+        PID pid = PIDs.get(pidString);
+
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        aclService.assertHasAccess("Insufficient permissions to assign thumbnail for " + pidString,
+                pid, principals, Permission.editDescription);
+
+        var object = repositoryObjectLoader.getRepositoryObject(pid);
+        if (!(object instanceof WorkObject)) {
```

Comment:
I'm not sure if this matches up with the functionality in the processor. It seems like that is expecting the id to be the file rather than the work. We could make it so that the user can pass in either the work or the file object's id when deleting the thumbnail. Or update the processor to expect a work idea for delete requests. Or make the processor flexible about what it receives.

Reply:
Thanks for catching that. I ended up changing it so that it expects the File ID in all cases.